src/message/models/messageManager.ts

```diff
@@ -25,22 +28,32 @@ export class MessageManager {
     return newMessage;
   }
 
-  public getMessages(params: IQueryModel): ILogObject[] {
+  public async getMessages(params: IQueryModel): Promise<ILogObject[]> {
     this.logger.info({ msg: 'getting filtered messages with query params: ', params });
 
     const { sessionId, severity, component, messageType } = params;
 
-    const allMessages: ILogObject[] = localMessagesStore;
+    const queryBuilder = messageLogsDataSource.getRepository(Message).createQueryBuilder('log');
 
-    const filteredMessages = allMessages.filter((instance) => {
-      if (severity != null && instance.severity !== severity) return false;
-      if (component != null && instance.component !== component) return false;
-      if (messageType != null && instance.messageType !== messageType) return false;
-      if (sessionId != null && instance.sessionId !== sessionId) return false;
-      return true;
-    });
+    if (severity != null) {
+      queryBuilder.andWhere('log.severity = :severity', { severity });
+    }
+
+    if (component != null) {
+      queryBuilder.andWhere('log.component = :component', { component });
```

Comment:
Check if you convert "queryBuilder.andWhere('log.messageType = :messageType', { messageType });" to a method that will be used by all enum types